ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java

```diff
@@ -155,6 +155,7 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
             blockParam.getBaseFeePerGas(),
             blockParam.getPrevRandao(),
             0,
+            blockParam.getExcessDataGas().orElse(null),
```

Comment:
Could including this cause an issue for EngineNewPayloadV2 creating blocks that have the excess data gas field in them?

Reply:
I ran a test using V2 endpoints on main vs this branch to confirm no V2 regression is introduced by when excessDataGas is null.
If excessDataGas is included in newPayload however, it would indeed change the block hash. I don't see this as an immediate issue but once shanghai is live, a CL's buggy 4844 code could cause besu to miscalculate the root hash.
The same problem exists at the moment for V1 -> V2 migration w.r.t the new withdrawals fields. 

I can't tell from the spec what the EL behaviour should be. Maybe we should throw an error when V2 is called with excessDataGas, need to confirm the spec or understand what other ELs are doing